abstract_domains/numerical/interval_domain.py

```diff
@@ -0,0 +1,307 @@
+        if not isinstance(other, self.__class__):
+            raise NotImplementedError("Incomparable types!")
+        if self.empty() or other.empty():
+            raise NotImplementedError("Empty intervals are not comparable!")
+        return self.upper <= other.lower
+
+    def __gt__(self, other):
+        if not isinstance(other, self.__class__):
+            raise NotImplementedError("Incomparable types!")
+        if self.empty() or other.empty():
+            raise NotImplementedError("Empty intervals are not comparable!")
+        return self.lower > other.upper
+
+    def __ge__(self, other):
+        if not isinstance(other, self.__class__):
+            raise NotImplementedError("Incomparable types!")
+        if self.empty() or other.empty():
+            raise NotImplementedError("Empty intervals are not comparable!")
+        return self.lower >= other.upper
+
+    def __hash__(self):
+        return hash(repr(self))
+
+    def __repr__(self):
+        if self.empty():
+            return "∅"
+        else:
+            return f"[{self.lower},{self.upper}]"
+
+    def add(self, other) -> 'Interval':
+        if self.empty() or other.empty():
+            return self.set_empty()
+        else:
+            self.interval = (self.lower + other.lower, self.upper + other.upper)
+            return self
+
+    def sub(self, other) -> 'Interval':
+        if self.empty() or other.empty():
+            return self.set_empty()
+        else:
+            self.interval = (self.lower - other.upper, self.upper - other.lower)
+            return self
+
+    def mult(self, other) -> 'Interval':
+        if self.empty() or other.empty():
+            return self.set_empty()
+        else:
+            comb = [self.lower * other.lower, self.lower * other.upper, self.upper * other.lower,
+                    self.upper * other.upper]
+            self.interval = (min(comb), max(comb))
+            return self
+
+    def negate(self) -> 'Interval':
+        if self.empty():
+            return self
+        else:
+            self.interval = (-self.upper, -self.lower)
+            return self
+
+
+class IntervalLattice(Interval, BottomMixin):
+    def __repr__(self):
+        if self.is_bottom():
+            return "⊥"
+        else:
+            return super().__repr__()
+
+    def top(self) -> 'IntervalLattice':
+        self.lower = -inf
+        self.upper = inf
+        return self
+
+    def is_top(self) -> bool:
+        return self._lower == -inf and self._upper == inf
+
+    def is_bottom(self) -> bool:
+        # we have to check if interval is empty, or got empty by an operation on this interval
+        if self.empty():
+            self.bottom()
+        return super().is_bottom()
+
+    def _less_equal(self, other: 'IntervalLattice') -> bool:
+        # NOTE: do not use less equal operator of plain interval since that has different semantics (every value in
+        # interval is less equal than any value in other interval)
```

Comment:
I would remove the less equal operator that you added to the Interval class and remove this comment as well.